cloud-sql/mysql/mysql2/connect-connector-auto-iam-authn.js

```diff
@@ -0,0 +1,45 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+'use strict';
+// [START cloud_sql_mysql_mysql2_connect_connector]
+const mysql = require('mysql2/promise');
+const {Connector} = require('@google-cloud/cloud-sql-connector');
+
+// connectWithConnectorAutoIAMAuthn initializes a connection pool for a Cloud SQL instance
+// of MySQL using the Cloud SQL Node.js Connector.
+const connectWithConnectorAutoIAMAuthn = async config => {
+  // Note: Saving credentials in environment variables is convenient, but not
+  // secure - consider a more secure solution such as
+  // Cloud Secret Manager (https://cloud.google.com/secret-manager) to help
+  // keep secrets safe.
+  const connector = new Connector();
+  const clientOpts = await connector.getOptions({
+    instanceConnectionName: process.env.INSTANCE_CONNECTION_NAME,
+    ipType: process.env.PRIVATE_IP,
+    authType: 'IAM',
+  });
+  const dbConfig = {
+    ...clientOpts,
+    user: process.env.IAM_DB_USER, // e.g. 'my-db-user'
+    database: process.env.DB_NAME, // e.g. 'my-database'
+    // ... Specify additional properties here.
+    ...config,
+  };
+  // Establish a connection to the database.
+  return mysql.createPool(dbConfig);
+};
+// [END cloud_sql_mysql_mysql2_connect_connector]
+module.exports = connectWithConnectorAutoIAMAuthn;
```

Comment:
we don't _need_ but I judged it better to follow the same pattern across all the examples, on top of that it makes it easy to actually swap them in case you want to test in the context of the app